### PR TITLE
net-dns/knot: version bump

### PR DESCRIPTION
**Pull request by nemunaire**
Hi!

This is the latest versions, following latest releases for knot.

### CODE REVIEW

File: net-dns/knot/knot-2.7.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	newinitd "${FILESDIR}/knot.init" knot
+	if use systemd; then
+		systemd_newunit "${FILESDIR}/knot-1.service" knot.service
+	fi
+
+	find "${D}" -name '*.la' -delete || die
+}
+
+pkg_postinst() {
+	enewgroup knot 53
```
Comment (juippis) [inline net-dns/knot/knot-2.7.8.ebuild]: Hey, this is now handled via methods explained by https://www.gentoo.org/glep/glep-0081.html
Please see examples from other similar ebuilds. 
https://github.com/gentoo/gentoo/tree/master/acct-group for reference. 
https://wiki.gentoo.org/wiki/Project:Quality_Assurance/UID_GID_Assignment for available GIDs. 

This also needs to be reviewed in the gentoo-dev mailing list, check the archives for example prodecure again. Rest looks good.

Comment (juippis) [reply net-dns/knot/knot-2.7.8.ebuild]: I see 53 is already 'reserved' for know :)

Same applies to creating users, and the 2nd ebuild btw.

**Comment by nemunaire** [comment]
Here are the changes I made for the two ebuilds:

```diff
--- a/net-dns/knot/knot-2.7.8.ebuild
+++ b/net-dns/knot/knot-2.7.8.ebuild
@@ -3,7 +3,7 @@
 
 EAPI=7

-inherit systemd user
+inherit systemd
 
 DESCRIPTION="High-performance authoritative-only DNS server"
 HOMEPAGE="https://www.knot-dns.cz/"
@@ -29,6 +29,8 @@ KNOT_MODULES=(
 IUSE="doc caps +fastparser idn +libidn2 systemd +utils ${KNOT_MODULES[@]}"
 
 RDEPEND="
+       acct-group/knot
+       acct-user/knot
        dev-db/lmdb
        dev-libs/libedit
        dev-libs/userspace-rcu:=
@@ -91,7 +93,6 @@ src_install() {
        default
        
        rmdir "${D}/var/run/${PN}" "${D}/var/run/" || die
-       keepdir /var/lib/${PN}

        newinitd "${FILESDIR}/knot.init" knot
        if use systemd; then
@@ -100,8 +101,3 @@ src_install() {

        find "${D}" -name '*.la' -delete || die
 }
-
-pkg_postinst() {
-       enewgroup knot 53
-       enewuser knot 53 -1 /var/lib/knot knot
-}
```

File: net-dns/knot/knot-2.7.8.ebuild (hunk 1 above)
Comment (nemunaire) [reply net-dns/knot/knot-2.7.8.ebuild]: Hmm, I just send the mail to gentoo-dev, but the operator of the list is zealous. I'll try to contact him...

Comment (juippis) [reply net-dns/knot/knot-2.7.8.ebuild]: I saw the message where you wanted to reserve UID/GID. If it doesnt receive replies in few days, lets move ahead.

Comment (nemunaire) [reply net-dns/knot/knot-2.7.8.ebuild]: Oh great, false positive alert then...

Understood.

**Comment by petteyg** [comment]
There are lots of DNS implementations. Consider that user 53 might better be a generic user "dns" rather than claimed by one single package.

**Comment by nemunaire** [comment]
@petteyg Please join the ongoing discussion on gentoo-dev: https://archives.gentoo.org/gentoo-dev/message/84f4a86f763e22c5aab9c9893ad4a3e9

Sure it can be anything else. The fact is that there is no generic dns user account (each DNS server has its own user: named (40), pdnsd (184), nsd (223), ...), and other distro use sometime the 53 UID/GID for something unrelated to DNS.

**Comment by nemunaire** [comment]
I just updated the PR, to follow advice received on gentoo-dev: 
https://archives.gentoo.org/gentoo-dev/message/286a4c441459ad9a5dada23c40f0b1e4

I think it's now time to merge!

**Comment by juippis** [comment]
Could you rebase on top of fresh master so we get a green light from the CI bot? :)

**Pull request closed without merging**